movement-test/src/main.c

```diff
@@ -0,0 +1,148 @@
+}
+
+// Update the tiles of the sprite to animate it.
+//
+// sprite_id: the id ("nb") of the sprite to update
+// anim:      pointer to the animation data
+// direction: direction of the animation (= offset of the animation in the global animation data)
+// frame:     the new frame of the animation that will be displayed
+//
+// Returns the next frame of the animation.
+UINT8 update_sprite_animation(UINT8 sprite_id, UINT8 *anim, UINT8 direction, UINT8 frame) {
+    UINT8 len = anim[direction];
+    UINT8 flip = anim[direction + 1 + frame]; // +1 : skip length
+    UINT8 tile_id = anim[direction + 1 + len + frame]; // +5: skip length+flip data
+
+    if (flip) {
+        flip_sprite_horiz(sprite_id);
+    } else {
+        unflip_sprite_horiz(sprite_id);
+    }
+
+    set_sprite_tile(sprite_id, tile_id);
+
+    return (frame + 1) % len;
+}
+
+void main(void) {
+    UINT8 keys = 0;
+    UINT8 frame_skip = 8;  // Update player's animation every 8 frame to
+                           // slow down the animation (8 frames = ~133 ms
+                           // between each animation frames)
+
+    // Initialize player's state
+    player_x = 80;
+    player_y = 72;
+    player_direction = PLAYER_DIRECTION_DOWN;
+    player_animation_frame = 0;
+    is_player_walking = 0;
+
+    // Load sprites' tiles in video memory
+    set_sprite_data(0, PLAYER_SPRITES_TILE_COUNT, PLAYER_SPRITES);
+
+    // Use 8x16 sprites
+    SPRITES_8x16;
+
+    // Makes sprites "layer" visible
+    SHOW_SPRITES;
+
+    // Init the sprite used for the player
+    move_sprite(PLAYER_SPRITE_ID, player_x, player_y);
+
+    // https://gbdev.gg8.se/wiki/articles/GBDK_set_sprite_prop
+    set_sprite_prop(PLAYER_SPRITE_ID, 0x00U);
+
+
+    while (1) {
+        // Wait for v-blank (screen refresh)
+        wait_vbl_done();
+
+        // Read joypad keys to know if the player is walking
+        // and in which direction
+        keys = joypad();
+        if (keys & J_UP) {
+            player_direction = PLAYER_DIRECTION_UP;
+            is_player_walking = 1;
+        } else if (keys & J_DOWN) {
+            player_direction = PLAYER_DIRECTION_DOWN;
+            is_player_walking = 1;
+        } else if (keys & J_LEFT) {
+            player_direction = PLAYER_DIRECTION_LEFT;
+            is_player_walking = 1;
+        } else if (keys & J_RIGHT) {
+            player_direction = PLAYER_DIRECTION_RIGHT;
+            is_player_walking = 1;
+        } else {
+            is_player_walking = 0;
+            frame_skip = 1;  // Force refresh of the animation frame
+        }
+
+        // Update the player position if it is walking
+        if (is_player_walking) {
+            if (player_direction == PLAYER_DIRECTION_RIGHT) player_x += 1;
+            else if (player_direction == PLAYER_DIRECTION_LEFT) player_x -= 1;
+            else if (player_direction == PLAYER_DIRECTION_UP) player_y -= 1;
+            else if (player_direction == PLAYER_DIRECTION_DOWN) player_y += 1;
```

Comment:
Should simplify this by introducing `dx` and `dy` and updating `player_x` and `player_y` with this